recipes/openimageio/all/conanfile.py

```diff
@@ -63,260 +64,248 @@ class OpenImageIOConan(ConanFile):
             self.requires("openjpeg/2.5.0")
         if self.options.with_openvdb:
             self.requires("openvdb/8.0.1")
         if self.options.with_ptex:
-            self.requires("ptex/2.4.0")
+            self.requires("ptex/2.4.2")
         if self.options.with_libwebp:
-            self.requires("libwebp/1.3.1")
+            self.requires("libwebp/1.3.2")
         # TODO: R3DSDK dependency
         # TODO: Nuke dependency
 
     def validate(self):
-        if self.settings.compiler.get_safe("cppstd"):
-            if Version(self.version) >= "2.3.0.0" or self.options.with_openvdb:
-                check_min_cppstd(self, 14)
-            else:
-                check_min_cppstd(self, 11)
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, 14)
         if is_msvc(self) and is_msvc_static_runtime(self) and self.options.shared:
             raise ConanInvalidConfiguration(
                 "Building shared library with static runtime is not supported!"
             )
 
-    def source(self):
-        files.get(
-            self,
-            **self.conan_data["sources"][self.version],
-            strip_root=True,
-            destination=self._source_subfolder
-        )
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
-    def _patch_sources(self):
-        files.apply_conandata_patches(self)
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
+    def generate(self):
+        tc = CMakeToolchain(self)
 
         # CMake options
-        cmake.definitions["CMAKE_DEBUG_POSTFIX"] = ""  # Needed for 2.3.x.x+ versions
-        cmake.definitions["OIIO_BUILD_TOOLS"] = True
-        cmake.definitions["OIIO_BUILD_TESTS"] = False
-        cmake.definitions["BUILD_DOCS"] = False
-        cmake.definitions["INSTALL_DOCS"] = False
-        cmake.definitions["INSTALL_FONTS"] = False
-        cmake.definitions["INSTALL_CMAKE_HELPER"] = False
-        cmake.definitions["EMBEDPLUGINS"] = True
-        cmake.definitions["USE_PYTHON"] = False
-        cmake.definitions["USE_EXTERNAL_PUGIXML"] = True
+        tc.variables["CMAKE_DEBUG_POSTFIX"] = ""  # Needed for 2.3.x.x+ versions
+        tc.variables["OIIO_BUILD_TOOLS"] = True
+        tc.variables["OIIO_BUILD_TESTS"] = False
+        tc.variables["BUILD_DOCS"] = False
+        tc.variables["INSTALL_DOCS"] = False
+        tc.variables["INSTALL_FONTS"] = False
+        tc.variables["INSTALL_CMAKE_HELPER"] = False
+        tc.variables["EMBEDPLUGINS"] = True
+        tc.variables["USE_PYTHON"] = False
+        tc.variables["USE_EXTERNAL_PUGIXML"] = True
+        tc.variables["BUILD_MISSING_FMT"] = False
+
+        # The tests automatically download test images via git which is avoided by the default not to build the tests
+        tc.variables["BUILD_TESTING"] = self.options.build_testing
```

Comment:
Should we maybe have `tools.build:skip_test` be reflected here?

Reply:
Maybe. I'm not fully sure how that works. I just did a best guess by looking at other related projects like openexr and the old conan 1 file. How would that best be done?